Makefile.work

```diff
@@ -153,6 +162,31 @@ ifneq ($(SONIC_DPKG_CACHE_SOURCE),)
 	DOCKER_RUN += -v "$(SONIC_DPKG_CACHE_SOURCE):/dpkg_cache:rw"
 endif
 
+
+# User name and tag for "docker-*" images created by native dockerd mode.
+ifeq ($(strip $(SONIC_CONFIG_USE_NATIVE_DOCKERD_FOR_BUILD)),y)
+DOCKER_USERNAME = -$(shell echo $(USER) | tr '[:upper:]' '[:lower:]')
+DOCKER_USERTAG = $(SLAVE_TAG)
+else
+DOCKER_USERNAME =
+DOCKER_USERTAG = latest
+endif
+
+# Define canned sequence to clean up Docker image cache.
+#   - These are the remnants from building the runtime Docker images using native (host) Docker daemon.
+#   - Image naming convention differs on a shared build system vs. non-shared.
+# $(docker-image-cleanup)
+ifeq ($(SONIC_CONFIG_USE_NATIVE_DOCKERD_FOR_BUILD),y)
+define docker-image-cleanup
+    @for i in $(shell docker images --quiet --filter 'dangling=true') ; do (docker rmi -f $$i &> /dev/null || true) ; done
+    @for i in $(shell docker images --quiet docker-*$(DOCKER_USERNAME):$(DOCKER_USERTAG)) ; do (docker rmi -f $$i &> /dev/null || true) ; done
```

Comment:
i think docker-*$(DOCKER_USERNAME) could match other dockers in the systems?

Reply:
Deletion is specific to a particular user with a combination of $(BRANCH)/$(USER)/$(PWD)/$(CONFIGURED_PLATFORM). The SLAVE_TAG hash is generated uniquely for Particular User, View, Configured platform and Branch name.